components/brave_wallet_ui/components/extension/buttons/nav-button/index.tsx

```diff
@@ -17,35 +23,78 @@ export type PanelButtonTypes =
   | 'sign'
   | 'reject'
 
-export interface Props {
+interface BaseProps {
   buttonType: PanelButtonTypes
   text: string | undefined
-  onSubmit: () => void
   disabled?: boolean
   needsTopMargin?: boolean
+  maxHeight?: string
+  minWidth?: string
 }
 
-export default class NavButton extends React.PureComponent<Props, {}> {
-  render () {
-    const { onSubmit, text, buttonType, disabled, needsTopMargin } = this.props
-    return (
-      <StyledButton
-        disabled={disabled}
-        buttonType={buttonType}
-        onClick={onSubmit}
-        addTopMargin={needsTopMargin && text ? text.length > 20 : false}
-      >
-        {buttonType === 'reject' &&
-          <RejectIcon />
-        }
-        {buttonType === 'sign' &&
-          <SignIcon />
-        }
-        {buttonType === 'confirm' &&
-          <ConfirmIcon />
-        }
-        <ButtonText buttonType={buttonType}>{text}</ButtonText>
-      </StyledButton>
-    )
+type ClickProps = (
+  {
+    onSubmit: () => void
+    url?: string
+  }
+  | {
+    url: string
+    onSubmit?: () => void
   }
+)
+
+export type Props = BaseProps & ClickProps
+
+export const NavButton: React.FC<Props> = ({
+  buttonType,
+  disabled,
+  maxHeight,
+  minWidth,
+  needsTopMargin,
+  onSubmit,
+  text,
+  url
+}) => {
+  // memos
+  const buttonContent = React.useMemo(() => {
+    return <>
+      {buttonType === 'reject' &&
+        <RejectIcon />
+      }
+      {buttonType === 'sign' &&
+        <SignIcon />
+      }
+      {buttonType === 'confirm' &&
+        <ConfirmIcon />
+      }
+      <ButtonText buttonType={buttonType}>{text}</ButtonText>
+    </>
+  }, [buttonType, text])
+
+  // render
+  return url ? (
+    <StyledLink
+      disabled={disabled}
+      buttonType={buttonType}
+      onClick={onSubmit}
+      to={url || ''}
+      maxHeight={maxHeight}
+      minWidth={minWidth}
+    >
+      {buttonContent}
+    </StyledLink>
+  ) : (
+    <StyledButton
```

Comment:
The nav button can now be rendered as an "a" tag

Reply:
@thypon  @kdenhartog  check for HTML injection attacks

Reply:
^ Not sure what I was thinking during the review to ask for this, but I don't see any concerns with HTML injection going on here. Looks fine to me